src/js/out/modules/node/http.js

```diff
@@ -834,27 +834,25 @@ class ClientRequest extends OutgoingMessage {
     return this.#protocol;
   }
   _write(chunk, encoding, callback) {
-    var body = this.#body;
-    if (!body) {
-      this.#body = chunk, callback();
+    if (!this.#bodyChunks) {
+      this.#bodyChunks = [chunk], callback();
       return;
     }
-    this.#body = body + chunk, callback();
+    this.#bodyChunks.push(chunk), callback();
   }
   _writev(chunks, callback) {
-    var body = this.#body;
-    if (!body) {
-      this.#body = chunks.join(), callback();
+    if (!this.#bodyChunks) {
+      this.#bodyChunks = chunks, callback();
       return;
     }
-    this.#body = body + chunks.join(), callback();
+    this.#bodyChunks.push(...chunks), callback();
   }
   _final(callback) {
     if (this.#finished = !0, this[kAbortController] = new AbortController, this[kAbortController].signal.addEventListener("abort", () => {
       this[kClearTimeout]();
     }), this.#signal?.aborted)
       this[kAbortController].abort();
-    var method = this.#method, body = this.#body;
+    var method = this.#method, body = Buffer.concat(this.#bodyChunks || []);
```

Comment:
nit: can we add a check for if the number of chunks was exactly 1 and if so, use the existing hcunk instead of creating a new one?

Reply:
Fixed in https://github.com/oven-sh/bun/pull/3946